Update laravel-auto-presenter dependency

philtweir opened the pull request:
Provide an acceptable version pair to allow use with later versions of mccool/laravel-auto-presenter. This allows timegrid/concierge to continue working with Laravel after 5.3.x (mccool/laravel-auto-presenter ~4.0 limits illuminate/view, and so Laravel, to 5.3.4 at most)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/timegridio/concierge/3)
<!-- Reviewable:end -->

philtweir committed:
Update laravel-auto-presenter dependency

Provide an acceptable version pair to allow use with later
mccool/laravel-auto-presenter (and so Laravel >=5.5)

alariva commented:
Thanks @philtweir 

Unfortunately, the build shows conflicts

https://travis-ci.org/timegridio/concierge/jobs/403778581

philtweir commented:
Thanks for the quick response @alariva - it looks like the PHP used for build is 5.6.32, whereas more recent versions require PHP7 - will see what options there are to have support for both.

philtweir committed:
fix logical OR syntax and provide composer.lock for PHP5.6

philtweir committed:
provide a set of dependencies that ensures both PHP5.6 and PHP7.2 work with the lock-file

philtweir committed:
Updated tests for Laravel 5.4/7.2

The changes here:
* use `assertDatabaseHas` instead of `seeInDatabase`
* force the Business model to report its actual connection name
  (testbench) with `getConnectionName` rather than being blank,
  as is default for a "newed" model even on non-blank connection,
  otherwise `contains` assertions will always fail

philtweir commented:
Scrutinzer is using PHP 7.2, while Travis is using PHP 5.6. As both are trying to test using dependencies from the same lock-file this needs a single resolved set that works for both. As far as I can tell, there is only one version of Laravel that works simultaneously with PHP 5.6 and PHP 7.2, which is Laravel 5.4.36 (possibly one or two other nearby patch numbers <36). While the current library code for Concierge works with both, there were a couple of updates required to the tests for L5.4 support, and a couple of composer.json hints required to get a set of dependency versions that works for both (note that composer needs to be run under PHP 5.6 when generating the lock-file, to ensure a vendors folder that works for both versions)
